letsencrypt/cli.py

```diff
@@ -901,28 +936,42 @@ def renew(config, unused_plugins):
             logger.warning("Renewal configuration file %s produced an "
                            "unexpected error: %s. Skipping.", renewal_file, e)
             logger.debug("Traceback was:\n%s", traceback.format_exc())
+            parse_failures.append(renewal_file)
             continue
 
         try:
-            if renewal_candidate is not None:
+            if renewal_candidate is None:
+                parse_failures.append(renewal_file)
+            else:
                 # _reconstitute succeeded in producing a RenewableCert, so we
                 # have something to work with from this particular config file.
 
                 # XXX: ensure that each call here replaces the previous one
                 zope.component.provideUtility(lineage_config)
-                print("Trying...")
-                # Because obtain_cert itself indirectly decides whether to renew
-                # or not, we couldn't currently make a UI/logging distinction at
-                # this stage to indicate whether renewal was actually attempted
-                # (or successful).
-                obtain_cert(lineage_config,
-                            plugins_disco.PluginsRegistry.find_all(),
-                            renewal_candidate)
+                # Although obtain_cert itself also indirectly decides
+                # whether to renew or not, we need to check at this
+                # stage in order to avoid claiming that renewal
+                # succeeded when it wasn't even attempted (since
+                # obtain_cert wouldn't raise an error in that case).
+                if _should_renew(lineage_config, renewal_candidate):
```

Comment:
Since you're actually not calling `obtain_cert` in the case where `_should_renew` returns `False`, we shoouuld at least be consistent.  But why not then remove the call to `_should_renew` inside `obtain_cert` in the `lineage != None` case, and simply set `action = renew` there?